plugin-build/build.gradle.kts

```diff
@@ -54,14 +54,9 @@ tasks.shadowJar.configure {
     relocate("org.jetbrains.kotlin.com.intellij", "com.intellij")
     mergeServiceFiles()
     dependencies {
-        exclude(dependency("org.jetbrains.intellij.deps:trove4j"))
-        exclude(dependency("org.jetbrains:annotations"))
-        exclude(dependency("org.jetbrains.kotlin:kotlin-compiler-embeddable"))
-        exclude(dependency("org.jetbrains.kotlin:kotlin-daemon-embeddable"))
-        exclude(dependency("org.jetbrains.kotlin:kotlin-reflect"))
-        exclude(dependency("org.jetbrains.kotlin:kotlin-script-runtime"))
-        exclude(dependency("org.jetbrains.kotlin:kotlin-stdlib"))
-        exclude(dependency("org.jetbrains.kotlin:kotlin-stdlib-common"))
+        include(dependency("io.gitlab.arturbosch.detekt:.*"))
+        include(dependency("io.github.detekt:.*"))
+        include(dependency("org.yaml:snakeyaml"))
```

Comment:
I imagine that this is here for some good reason but `compileOnly` shouldn't be enough?

Reply:
Shadow plugin will include dependencies from `runtimeClasspath` configuration by default, but that means things like `kotlin-compiler-embeddable` will be included which isn't necessary because it's provided as part of the runtime environment anyway. They were excluded before, but it didn't exclude everything that wasn't required. It's best to include just what's required.